### PR TITLE
RF: Delegate isotropic smoothing to nistats

### PR DESCRIPTION
**Pull request by kfinc**
This PR delegates smoothing to Nistats. After making the changes we came into multiprocessing error. Any idea why this may be happening? Thanks in advance for any help!

<details><summary>Expand to see stack trace</summary>
<p>

```python
	 [Node] Running "_l1_model7" ("fitlins.interfaces.nistats.FirstLevelModel")
190227-19:32:39,265 nipype.workflow INFO:
	 [MultiProc] Running 4 tasks, and 4 jobs ready. Free memory (GB): 13.19/13.99, Free processors: 0/4.
                     Currently running:
                       * _l1_model7
                       * _l1_model6
                       * _l1_model5
                       * _l1_model4
exception calling callback for <Future at 0x7f1dcf8fec18 state=finished raised BrokenProcessPool>
Traceback (most recent call last):
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 324, in _invoke_callbacks
    callback(self)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/site-packages/nipype/pipeline/plugins/multiproc.py", line 151, in _async_callback
    result = args.result()
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
exception calling callback for <Future at 0x7f1dcf052a58 state=finished raised BrokenProcessPool>
Traceback (most recent call last):
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 324, in _invoke_callbacks
    callback(self)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/site-packages/nipype/pipeline/plugins/multiproc.py", line 151, in _async_callback
    result = args.result()
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
exception calling callback for <Future at 0x7f1dcf0b0c18 state=finished raised BrokenProcessPool>
Traceback (most recent call last):
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 324, in _invoke_callbacks
    callback(self)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/site-packages/nipype/pipeline/plugins/multiproc.py", line 151, in _async_callback
    result = args.result()
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
exception calling callback for <Future at 0x7f1dcf060550 state=finished raised BrokenProcessPool>
Traceback (most recent call last):
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 324, in _invoke_callbacks
    callback(self)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/site-packages/nipype/pipeline/plugins/multiproc.py", line 151, in _async_callback
    result = args.result()
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/opt/miniconda-latest/envs/neuro/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
```

### CODE REVIEW

**Comment by oesteban** [comment]
You are running out of memory. Try not to parallelize 4 models at a time or increase the RAM available to the docker engine.

File: fitlins/workflows/base.py [hunk 1]
```diff
@@ -60,13 +60,12 @@ def init_fitlins_wf(bids_dir, derivatives, out_dir, space, desc=None,
...
-            name='smoother')
+        smoothing_fwhm = float(smoothing_params[1])
+    else:
+        smoothing_fwhm = None
 
     l1_model = pe.MapNode(
-        FirstLevelModel(),
+        FirstLevelModel(smoothing_fwhm=smoothing_fwhm),
         iterfield=['session_info', 'contrast_info', 'bold_file', 'mask_file'],
         name='l1_model')
```
Comment (effigies) [inline fitlins/workflows/base.py]:
```Python
if smoothing:
    l1_model.inputs.smoothing_fwhm = smoothing_fwhm
```

**Comment by effigies** [comment]
LGTM. Is this still a WIP?

**Comment by kfinc** [comment]
Everything works fine, so I think it's no longer WIP. Thanks everyone for help!

**Pull request merged**